### DISCUSSION TITLE
Is it possible to use a PAT token in the cli?

### DISCUSSION THREAD

**Discussion by ostiwe** [Infisical/infisical#1189] [Q&A]
Hi, I'm writing automatic creation of environments when branches are deployed. I need to create new environments, if there are none, then ok, I use the API in the script:
```sh
curl --location "$API_HOST/v2/workspace/$WORKSPACE/environments" \
            --header "X-API-Key: $API_KEY" \
            --header 'Content-Type: application/json' \
            --data "{
              \"environmentName\": \"$CI_COMMIT_BRANCH\",
              \"environmentSlug\": \"$TARGET_ENV\"
           }"
```

This is where the problems begin, I can't create a service token, since there is no API method for this, and as I understand it, it is impossible to use PAT in the cli.

Accordingly, the question is: what should I do in this situation? Or is it possible to use PAT in the cli somehow?

**Comment by maidul98** [comment]
@ostiwe this will soon be solved when we release a new machine identity method! Look out for it end of this week
